packages/config/src/index.ts

```diff
@@ -40,10 +41,11 @@ class TruffleConfig {
     );
   }
 
-  public eventManagerOptions(config: TruffleConfig): any {
-    let muteLogging;
-    const { quiet, logger, subscribers } = config;
-    return { logger, quiet, subscribers };
+  private eventManagerOptions(
+    options: TruffleConfig | { quiet?: boolean, logger?: any, subscribers?: any[] }
+  ): any {
```

Comment:
This return type should I think get a real type rather than being `any`, I think.  The uses of `any` for `logger` and `subscribers` I think are fine.  But the return type shouldn't just be `any`.  (You'll likely want to make an actual type definition here.)

(There's actually probably a better way to type this that @gnidan might know, but I don't care enough to be that picky. :P )

Reply:
I guess the fancy way to do this would be with the `Pick` type: https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

But as mentioned I won't insist on that :P

Reply:
Oh yeah that is probably the way to do it!

Reply:
Although, is there a way to pick where all properties are optional? Would that be
```
Partial<Pick<TruffleConfig, "quiet" | "logger" | "subscribers">>
```?

Reply:
I thought that would be the return type as well but TS didn't like it for some reason. So I just left it as `Partial<TruffleConfig>`.